soda_machine.rb

```diff
@@ -7,12 +7,21 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.count
   end
 
   def find_soda(soda_brand)
+    @sodas.find{|soda|
+      soda.brand == soda_brand
+    }
   end
 
   def sell(soda_brand)
+    soda = find_soda(soda_brand)
```

Comment:
one way to refactor could be
 ```
return nil if find_soda(soda_brand) == nil 
```

and then if it isn't, store the result in a variable so that you don't store nil unnecessarily in a variable.